scripts/build/Jenkins/Jenkinsfile

```diff
@@ -468,7 +468,18 @@ def CreateSingleNode(Map pipelineConfig, def platform, def build_job, Map envVar
                     if(build_job.value.steps) { //this is a pipe with many steps so create all the build stages
                         build_job.value.steps.each { build_step ->
                             build_job_name = build_step
-                            CreateBuildStage(pipelineConfig,  platform.key, build_step, envVars).call()
+                            envVars = GetBuildEnvVars(platform.value.PIPELINE_ENV ?: EMPTY_JSON, platform.value.build_types[build_step].PIPELINE_ENV ?: EMPTY_JSON, pipelineName)
+                            try {
+                                CreateBuildStage(pipelineConfig,  platform.key, build_step, envVars).call()
+                            }
+                            catch (Exception e) {
+                                if (envVars['NONBLOCKING_STEP']?.toBoolean()) {
+                                    unstable(message: "Build step ${build_step} failed but it's a non-blocking step in build job ${build_job.key}")
+                                }
+                                else {
+                                    error "FAILURE: ${e}"
```

Comment:
Should we return early here if it's a blocking step?

Reply:
Yeah, the error command in else block is to return early if it's a blocking step.